src/java/org/apache/cassandra/index/sai/utils/IndexFileUtils.java

```diff
@@ -132,7 +133,7 @@ public void recalculateChecksum() throws IOException
             if (!file.exists())
```

Comment:
Flush only creates single segment, so the previous segment's file (eg. terms_data, etc) doesn't exist. So NSS on writer won't hit this issue in the next line.